core/src/test/java/org/mobilitydata/gtfsvalidator/table/AnyTableLoaderTest.java

```diff
@@ -0,0 +1,239 @@
+        .contains(new MissingRequiredColumnNotice("filename", "route_id"));
+    assertThat(loadedContainer).isEqualTo(mockContainer);
+  }
+
+  @Test
+  public void invalidRowLengthNotice() {
+    var testTableDescriptor = spy(new GtfsTestTableDescriptor());
+    GtfsTableContainer mockContainer = mock(GtfsTableContainer.class);
+    when(testTableDescriptor.createContainerForInvalidStatus(
+            GtfsTableContainer.TableStatus.UNPARSABLE_ROWS))
+        .thenReturn(mockContainer);
+    ValidatorProvider validatorProvider =
+        spy(new DefaultValidatorProvider(validationContext, ValidatorLoader.createEmpty()));
+    NoticeContainer loaderNotices = new NoticeContainer();
+    InputStream inputStream = toInputStream("id,code\n" + "s1\n");
+
+    var loadedContainer =
+        AnyTableLoader.load(testTableDescriptor, validatorProvider, inputStream, loaderNotices);
+
+    assertThat(loaderNotices.hasValidationErrors()).isTrue();
+    assertThat(loaderNotices.getValidationNotices())
+        .containsExactly(new InvalidRowLengthNotice("filename.txt", 2, 1, 2));
+    assertThat(loadedContainer).isEqualTo(mockContainer);
+  }
+
+  @Test
+  public void missingRequiredColumn() {
+    var testTableDescriptor = spy(new GtfsTestTableDescriptor());
+    when(testTableDescriptor.getColumns())
+        .thenReturn(
+            ImmutableList.of(
+                GtfsColumnDescriptor.builder()
+                    .setColumnName(GtfsTestEntity.ID_FIELD_NAME)
+                    .setHeaderRequired(true)
+                    .setFieldLevel(FieldLevelEnum.REQUIRED)
+                    .setIsMixedCase(false)
+                    .setIsCached(false)
+                    .build(),
+                GtfsColumnDescriptor.builder()
+                    .setColumnName(GtfsTestEntity.CODE_FIELD_NAME)
+                    .setHeaderRequired(true)
+                    .setFieldLevel(FieldLevelEnum.REQUIRED)
+                    .setIsMixedCase(false)
+                    .setIsCached(false)
+                    .build()));
+    GtfsTableContainer mockContainer = mock(GtfsTableContainer.class);
+    when(testTableDescriptor.createContainerForInvalidStatus(
+            GtfsTableContainer.TableStatus.INVALID_HEADERS))
+        .thenReturn(mockContainer);
+    ValidatorProvider validatorProvider =
+        spy(new DefaultValidatorProvider(validationContext, ValidatorLoader.createEmpty()));
+    NoticeContainer loaderNotices = new NoticeContainer();
+    InputStream inputStream = toInputStream("id\n" + "s1 \n");
+
+    var loadedContainer =
+        AnyTableLoader.load(testTableDescriptor, validatorProvider, inputStream, loaderNotices);
+
+    assertThat(loaderNotices.hasValidationErrors()).isTrue();
+    assertThat(loaderNotices.getValidationNotices())
+        .containsExactly(new MissingRequiredColumnNotice("filename.txt", "code"));
+    assertThat(loadedContainer).isEqualTo(mockContainer);
+  }
+
+  @Test
+  public void parsableTableRows() {
```

Comment:
Ditto comment above about `DefaultTableHeaderValidator`.  I think if you want to mock `ValidatorProvider` entirely and verify that different types of validators are created and called (like you do with the mocked `GtfsTestFileValidator` in this test), that's reasonable but I wouldn't mix in the behavior of `DefaultValidatorProvider`, which could change independently of `AnyTableLoader`.

Reply:
Removed all dependencies with DefaultTableHeaderValidator and DefaultValidatorProvider.